### PR TITLE
Issue #2633: Create a User Interface for editing an existing managed …

### PR DESCRIPTION
**Pull request by jenlampton**
Fixes https://github.com/backdrop/backdrop-issues/issues/2796

Replaces https://github.com/backdrop/backdrop/pull/1997

### CODE REVIEW

File: core/modules/file/config/views.view.file_admin.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
     "name": "file_admin",
     "description": "Administrative listing for managing files.",
     "module": "file",
-    "storage": 4,
+    "storage": 2,
```
Comment (quicksketch) [inline core/modules/file/config/views.view.file_admin.json]: This needs to stay `4`, indicating the default state.

File: core/modules/file/file.install [hunk 2]
```diff
@@ -648,3 +648,91 @@ function file_update_1000() {
...
+        'word_boundary' => TRUE,
+        'ellipsis' => TRUE,
+        'more_link' => 0,
+        'more_link_text' => '',
+        'more_link_path' => '',
+        'strip_tags' => 0,
+        'trim' => 0,
+        'preserve_tags' => '',
+        'html' => 0,
+      ),
```
Comment (quicksketch) [inline core/modules/file/file.install]: I usually exclude the `alter` array if all the options are the default, then they will be repopulated by Views on the loading of the view.

File: core/modules/file/file.install [hunk 3]
```diff
@@ -648,3 +648,91 @@ function file_update_1000() {
...
+    $dropbutton['fields'] = array('edit' => 'edit') + $dropbutton['fields'];
+
+    // Add the dropbutton field back to the view.
+    $fields['dropbutton'] = $dropbutton;
+
+    // Save the view.
+    $config->set('display.default.display_options.fields', $fields);
+    $config->save();
+
+    cache_clear_all();
```
Comment (quicksketch) [inline core/modules/file/file.install]: cache_clear_all() is not needed, this is done automatically after update.php finishes.

File: core/modules/file/file.pages.inc [hunk 4]
```diff
@@ -1,9 +1,388 @@
...
 
+/**
+ * Page callback: Form constructor for the file edit form.
+ *
+ * @param object $file
+ *   A file object from file_load().
+ */
+function file_edit_form($form, &$form_state, $file) {
+  // @todo Add type w/ https://github.com/backdrop/backdrop-issues/issues/2632
+  backdrop_set_title(t('<em>Manage file</em> @title', array('@title' => $file->filename)), PASS_THROUGH);
```
Comment (quicksketch) [inline core/modules/file/file.pages.inc]: This title seems backwards. Don't we normally show the title in italics and the verb (Manage, Edit, Delete, etc) in the non-italicized font? `t('Manage file %name', array('%name' => $file->filename))`

File: core/modules/file/file.pages.inc [hunk 5]
```diff
@@ -1,9 +1,388 @@
...
+
+    $args['@orphaned'] = file_uri_target($orphaned_uri);
+    watchdog('file', 'File: deleted orphaned file @orphaned for %title.', $args);
+    backdrop_set_message(t('The replaced @type @orphaned has been deleted.', $args));
+  }
+
+  $form_state['redirect'] = 'admin/content/files';
+
+  // Clear the page and block caches.
+  cache_clear_all();
```
Comment (quicksketch) [inline core/modules/file/file.pages.inc]: Recommend we remove this line. In D7, global `cache_clear_all()` commands were common on node and comment save, but we have remove those for performance in Backdrop.

File: core/modules/image/image.module [hunk 6]
```diff
@@ -468,7 +468,7 @@ function image_field_update_instance($instance, $prior_instance) {
  * Clear cached versions of a specific file in all styles.
  *
  * @param $path
- *   The Backdrop file path to the original image.
+ *   The uri or path to the original image.
```
Comment (quicksketch) [inline core/modules/image/image.module]: Let's capitalize URI here.

File: core/modules/file/file.pages.inc [hunk 7]
```diff
@@ -1,9 +1,388 @@
...
+    $validators += $options['upload_validators'];
+  }
+
+  return $validators;
+}
+
+/**
+ * Pre-render callback for adding validation descriptions to file upload fields.
+ */
+function file_upload_validators_pre_render($element) {
```
Comment (quicksketch) [inline core/modules/file/file.pages.inc]: Lets move this function into file.module, as it has generic usefulness.

**Review by quicksketch — CHANGES_REQUESTED** [review]

File: core/modules/file/file.install (hunk 2 above)
Comment (jenlampton) [reply core/modules/file/file.install]: nice, done.

File: core/modules/file/file.install (hunk 3 above)
Comment (jenlampton) [reply core/modules/file/file.install]: removed.

File: core/modules/file/file.pages.inc (hunk 7 above)
Comment (jenlampton) [reply core/modules/file/file.pages.inc]: done.

File: core/modules/image/image.module (hunk 6 above)
Comment (jenlampton) [reply core/modules/image/image.module]: done.

**Comment by quicksketch** [comment]
Looks great! All the new tests are passing with the new terminology, and the update hook appears to work well. Merged into 1.x for 1.9.0.

**Pull request merged**